src/mlpack/methods/emst/dtb_impl.hpp

```diff
@@ -103,6 +103,56 @@ template<
     template<typename TreeMetricType,
              typename TreeStatType,
              typename TreeMatType> class TreeType>
+DualTreeBoruvka<MetricType, MatType, TreeType>::DualTreeBoruvka(
+    Tree& tree,
+    const MetricType metric) :
+    tree(new Tree(tree)),
+    data(&this->tree->Dataset()),
+    ownTree(true),
+    naive(false),
+    connections(data.n_cols),
+    totalDist(0.0),
+    metric(metric)
```

Comment:
The problem, I believe, is the line `data(&this->tree->Dataset())`.  `this->tree->Dataset()` is already a reference, so there's no need to take the address of it.